src/target/riscv/riscv-013.c

```diff
@@ -2237,10 +2237,18 @@ static int sample_memory_bus_v1(struct target *target,
 		if (result != ERROR_OK)
 			return result;
 
+		/* Discard the batch when we encounter a busy state. It's too
+		 * much hassle to try to recover partial data. We'll try again
+		 * with a larger delay. */
+		unsigned int sbcs_read_op = riscv_batch_get_dmi_read_op(batch, sbcs_key);
+		if (sbcs_read_op == DTM_DMI_OP_BUSY) {
+			increase_dmi_busy_delay(target);
+			continue;
+		}
+
 		uint32_t sbcs_read = riscv_batch_get_dmi_read_data(batch, sbcs_key);
-		if (get_field(sbcs_read, DM_SBCS_SBBUSYERROR)) {
-			/* Discard this batch (too much hassle to try to recover partial
-			 * data) and try again with a larger delay. */
+		if (get_field(sbcs_read, DM_SBCS_SBBUSYERROR) ||
+				sbcs_read_op == DTM_DMI_OP_BUSY) {
```

Comment:
The second part of the condition - `sbcs_read_op == DTM_DMI_OP_BUSY` - will never be `true` because there is already an identical check above on line 2244.

Could you please double-check the code. (Maybe you intended some other condition here?)

Reply:
Good catch. That was left over from an earlier attempt where I tried to combine both busy checks. Then I made the new one on 2244 and didn't fix this.